wasm/src/lib.rs

```diff
@@ -64,6 +62,74 @@ unsafe extern "C" fn _iroha_wasm_dealloc(offset: *mut u8, len: usize) {
     let _box = Box::from_raw(core::slice::from_raw_parts_mut(offset, len));
 }
 
+/// Marks a type whose values can be queried from the host environment and passed
+/// as arguments to `entrypoint` of a smart contract
+pub trait QueryHost {
```

Comment:
```suggestion
pub trait QueryHost: sealed::Sealed {
```

I think we should prevent users to write impls for this trait for their types